plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/CodeTemplates.java

```diff
@@ -138,96 +142,83 @@ private static IFile materialize(IProject project, AppEngineProjectConfig config
-        if (Objects.equal(AppEngineRuntime.STANDARD_JAVA_7.getId(), config.getRuntimeId())) {
-          mavenCoordinates.put("servletVersion", "2.5"); //$NON-NLS-1$ //$NON-NLS-2$
-          mavenCoordinates.put("compilerVersion", "1.7"); //$NON-NLS-1$ //$NON-NLS-2$
-        } else {
-          mavenCoordinates.put("servletVersion", "3.1"); //$NON-NLS-1$ //$NON-NLS-2$
-          mavenCoordinates.put("compilerVersion", "1.8"); //$NON-NLS-1$ //$NON-NLS-2$
-        }
-        createChildFile("pom.xml", Templates.POM_XML_STANDARD_TEMPLATE, //$NON-NLS-1$
-            project, mavenCoordinates, subMonitor.newChild(5));
-      } else {
-        createChildFile("pom.xml", Templates.POM_XML_FLEX_TEMPLATE, //$NON-NLS-1$
-            project, mavenCoordinates, subMonitor.newChild(5));
-      }
-    }
-
-    return hello;
   }
 
-  private static IFolder createFoldersForPackage(IFolder parentFolder,
-                                                 String packageName,
-                                                 SubMonitor subMonitor) throws CoreException {
-    IFolder folder = parentFolder;
-    if (packageName != null && !packageName.isEmpty()) {
-      String[] packages = packageName.split("\\.");  //$NON-NLS-1$
-      subMonitor.setWorkRemaining(packages.length);
-      for (int i = 0; i < packages.length; i++) {
-        folder = createChildFolder(packages[i], folder, subMonitor.newChild(1));
+  private static void createPomXml(IProject project, AppEngineProjectConfig config,
+      boolean isStandardProject, IProgressMonitor monitor) throws CoreException {
+    Map<String, String> properties = new HashMap<>();
+    properties.put("projectGroupId", config.getMavenGroupId()); //$NON-NLS-1$
+    properties.put("projectArtifactId", config.getMavenArtifactId()); //$NON-NLS-1$
+    properties.put("projectVersion", config.getMavenVersion()); //$NON-NLS-1$
+
+    String template;
+    if (isStandardProject) {
+      if (Objects.equal(AppEngineRuntime.STANDARD_JAVA_7.getId(), config.getRuntimeId())) {
+        properties.put("servletVersion", "2.5"); //$NON-NLS-1$ //$NON-NLS-2$
+        properties.put("compilerVersion", "1.7"); //$NON-NLS-1$ //$NON-NLS-2$
+      } else {
+        properties.put("servletVersion", "3.1"); //$NON-NLS-1$ //$NON-NLS-2$
+        properties.put("compilerVersion", "1.8"); //$NON-NLS-1$ //$NON-NLS-2$
       }
+      template = Templates.POM_XML_STANDARD_TEMPLATE;
+    } else {
+      template = Templates.POM_XML_FLEX_TEMPLATE;
     }
-    return folder;
-  }
 
-  @VisibleForTesting
-  static IFolder createChildFolder(String name, IFolder parent, IProgressMonitor monitor)
-      throws CoreException {
-    monitor.subTask("Creating folder " + name);
-
-    boolean force = true;
-    boolean local = true;
-    IFolder child = parent.getFolder(name);
-    if (!child.exists()) {
-      child.create(force, local, monitor);
-    }
-    return child;
+    createChildFile("pom.xml", template, project, properties, monitor); //$NON-NLS-1$
```

Comment:
You might instead remove the template variable and move this call into the if-else blocks.

Reply:
Done.